src/tla/__rewire_sequences_ext_in_apalache.tla

```diff
@@ -0,0 +1,470 @@
+    IN
+    \* prune the longer seq
+    SubSeq(__raw_seq, 1, __next_len)
+  IN
+  __ApalacheFoldSet(__insert_sorted, <<>>, S)
+
+(**
+ * TupleOf(s, 3) = s \X s \X s
+ *
+ * This operator is not well-typed.
+ * Hence, we assign a reasonable type signature.
+ *
+ * @type: (Set(a), Int) => (Int -> Set(a));
+ *)
+TupleOf(set, n) == 
+  __NotSupportedByModelChecker("TupleOf")
+
+(**
+ * All sequences up to length n with all elements in set.  Includes empty
+ * sequence.
+ *
+ * Apalache does not support this operator. Try Apalache!Gen instead.
+ *
+ * @type: (Set(a), Int) => Set(Seq(a));
+ *)
+SeqOf(set, n) == 
+  __NotSupportedByModelChecker("SeqOf")
+
+(**
+ * An alias for SeqOf to make the connection to Sequences!Seq, which is
+ * the unbounded version of BoundedSeq.
+ *
+ * Apalache does not support this operator. Try Apalache!Gen instead.
+ *
+ * @type: (Set(a), Int) => Set(Seq(a));
+ *)
+BoundedSeq(S, n) ==
+  __NotSupportedByModelChecker("SeqOf")
+
+(**
+ * TRUE iff the element e \in ToSet(s).
+ *
+ * @type: (Seq(a), a) => Bool;
+ *)
+Contains(s, e) ==
+  \E i \in DOMAIN s:
+    s[i] = e
+
+(**
+ * Reverse the given sequence s:  Let l be Len(s) (length of s).
+ * Equals a sequence s.t. << s[l], s[l - 1], ..., s[1]>>.
+ *
+ * @type: Seq(a) => Seq(a);
+ *)
+Reverse(s) ==
+  LET __s_len == Len(s) IN
+  LET __get_ith(i) == s[__s_len - i + 1] IN
+  SubSeq(__ApalacheMkSeq(__ApalacheSeqCapacity(s), __get_ith), 1, __s_len)
+
+(**
+ * The sequence s with e removed or s iff e \notin Range(s).
+ *
+ * @type: (Seq(a), a) => Seq(a);
+ *)
+Remove(seq, e) ==
+    LET __append_if_eq(__res, __t) ==
+        IF __t /= e
+        THEN Append(__res, __t)
+        ELSE __res
+    IN
+    __ApalacheFoldSeq(__append_if_eq, <<>>, seq)
+
+(**
+ * Equals the sequence s except that all occurrences of element old are
+ * replaced with the element new.
+ *
+ * @type: (Seq(a), a, a) => Seq(a);
+ *)
+ReplaceAll(seq, old, new) ==
+  LET __copy_or_set(__i) ==
+    IF seq[__i] = old THEN new ELSE seq[__i]
+  IN
+  SubSeq(__ApalacheMkSeq(__ApalacheSeqCapacity(seq), __copy_or_set),
+        1, Len(seq))
+  
+(**  
+ * Inserts element e at the position i moving the original element to i+1
+ * and so on.  In other words, a sequence t s.t.:
+ *   /\ Len(t) = Len(s) + 1
+ *   /\ t[i] = e
+ *   /\ \A j \in 1..(i - 1): t[j] = s[j]
+ *   /\ \A k \in (i + 1)..Len(s): t[k + 1] = s[k]
+ *
+ * @type: (Seq(a), Int, a) => Seq(a);
+ *)
+InsertAt(seq, k, e) ==
+  LET __copy_or_set(__i) ==
+    IF __i = k
+    THEN e
+    ELSE IF __i < k
+         THEN seq[__i]
+         ELSE seq[__i - 1]
+  IN
+  SubSeq(__ApalacheMkSeq(__ApalacheSeqCapacity(seq) + 1, __copy_or_set),
+        1, Len(seq) + 1)
+
+(**
+ * Replaces the element at position i with the element e.
+ *
+ * @type: (Seq(a), Int, a) => Seq(a);
+ *)
+ReplaceAt(s, i, e) ==
+  [s EXCEPT ![i] = e]  
+
+(**
+ * Replaces the element at position i shortening the length of s by one.
```

Comment:
```suggestion
 * Removes the element at position i shortening the length of s by one.
```
I know you want to keep this in sync with the community modules, but this is really confusing.